okhttp/src/commonJvmAndroid/kotlin/okhttp3/OkHttpClient.kt

```diff
@@ -359,7 +367,24 @@ open class OkHttpClient internal constructor(
   }
 
   /** Prepares the [request] to be executed at some point in the future. */
-  override fun newCall(request: Request): Call = RealCall(this, request, forWebSocket = false)
+  override fun newCall(request: Request): Call {
+    if (callDecorators.isNotEmpty()) {
+      return DecoratedCallFactory().newCall(request)
+    }
+
+    return RealCall(this, request, forWebSocket = false)
+  }
+
+  private inner class DecoratedCallFactory : Call.Factory {
+    var index: Int = 0
+
+    override fun newCall(request: Request): Call =
+      if (index > callDecorators.lastIndex) {
+        RealCall(this@OkHttpClient, request, forWebSocket = false)
+      } else {
+        callDecorators[index++].newCall(this, request)
```

Comment:
This mutability might be an issue if someone implements a decorator that the chain twice.